src/System.CommandLine.Tests/CustomTokenReplacerTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.CommandLine.Builder;
+using System.CommandLine.Parsing;
+using FluentAssertions;
+using Xunit;
+
+namespace System.CommandLine.Tests;
+
+public class CustomTokenReplacerTests
+{
+    [Fact]
+    public void Custom_token_replacer_can_expand_argument_values()
+    {
+        var argument = new Argument<int>();
+
+        var command = new RootCommand { argument };
+
+        var parser = new CommandLineBuilder(command)
+                     .UseTokenReplacer((string tokenToReplace, out IReadOnlyList<string> tokens, out string message) =>
```

Comment:
Should this test also assert on the value of `tokenToReplace`? I assume in this situation it would be "@interpolate-me" or would it be "interpolate-me"?

Reply:
I'll add a test for this. The received `tokenToReplace` will not include the `@`. Open to changing that. My assumption was that this should be transparently removed, sort of like quotation marks.

Reply:
That sounds good to me